### ISSUE TITLE
Second to last login command 

### ISSUE THREAD

**Issue by Klaranth** [pandorabox-io/in-game#207]
Hedgehog ; SX ; Test_User
Create a command that can work as a second to last login .... so we can find out when some folks that just arrived on server have been here last prior to that.

If doing that then better would be to just include x number of last login times into /last-login command instead of adding another command

I think currently there's technical limitation however, usable data for login history is not available (well it is but I think not in usable form for mods). Command itself would be very simple but decision about storing login history should be made before it is possible to add actual functionality for displaying login times.

You could use mod storage for a table with names as indexes and times for the value... actually two tables, one for current, one for time before, and use register_on_player_join to switch them all

I would rather go with nested table, root level hashed by player name and below that list of login times. Mostly because that removes limitations from data storage and moves possible limitations to functionality implementation.

There basically current login time is logintimes["playername"][#logintimes["playername"]] and previous is logintimes["playername"][#logintimes["playername"]-1] and so on. Actual implementation can limit number of stored login times or can keep all login times.

**Comment by SwissalpS** [comment]
just save last logout time and add command /last-logout <nick>
Players could still build their own luac or out of game trackers using web-api, but when lazy use chatcommand.

**Comment by S-S-X** [comment]
> just save last logout time and add command /last-logout nick Players could still build their own luac or out of game trackers using web-api, but when lazy use chatcommand.

It is a lot harder to make `/last-logout` reliable, requires tracking for logins and cleanup on every server start to handle cases where players are logged out because of server crash.

Web API does not give any better information unless you keep polling it constantly and saving login times into your own database.

**Comment by S-S-X** [comment]
> Would it be possible to store the last login time when a player joins, before it gets updated? That might work for a `/second-last-login` command...

I would rather go with proper login history database for few reasons:
* Does not depend on actual command, anything goes as command just reads database.
* Stores as many login times you want, not lower or upper limits.
* Simple records with join hook, when engine updates last login does not matter.
* Allows finding significant login times filtering out multiple logins caused by bad network connectivity.
* Being flexible from beginning allows future updates for behavior and functionality.
* Wire with `/second-last-login` or integrate with `/last-login` is just UI decision and completely decoupled from bookkeeping.

This list is meant to represent concept instead of implementation details even while I did mention possible implementation in issue description. Basic considerations are: simple and extensible implementation without locking records for single use case.

For actual command / UI anything goes but I would still prefer not adding more commands for basic functionality like this because additional command in my opinion is not really needed if you can just extend core command `/last-login`.

**Comment by SwissalpS** [comment]
so, `/last-login <nick> [<n entry>]` where n defaults to 1 == last login; 2 == second to last; 3 == third last; etc?

**Comment by S-S-X** [comment]
> so, `/last-login <nick> [<n entry>]` where n defaults to 1 == last login; 2 == second to last; 3 == third last; etc?

Yes and no, I mean possibly. Example of chat command implementations that should work without touching database used for login history or method used to record login times:

```
/last-login
You logged in 2021-01-01T00:11:11Z
You logged in 2022-02-02T00:22:22Z
You logged in 2023-03-03T00:33:33Z
Your last login time was 2024-04-04T00:44:44Z.

/last-login nick
nick logged in 2021-01-01T00:11:11Z
nick logged in 2022-02-02T00:22:22Z
nick logged in 2023-03-03T00:33:33Z
nick's last login time was 2024-04-04T00:44:44Z.

/last-login nick 2
nick logged in 2023-03-03T00:33:33Z

/second-last-login
You logged in 2023-03-03T00:33:33Z

/second-last-login nick
nick logged in 2023-03-03T00:33:33Z
```
